src/molecules/Chip/InteractiveChip.tsx

```diff
@@ -3,51 +3,46 @@ import { forwardRef, HTMLAttributes, KeyboardEvent } from 'react';
 import { Icon } from '@equinor/eds-core-react';
 import { close } from '@equinor/eds-icons';
 
-import { BaseChipProps } from 'src/molecules/Chip/Chip';
+import {
+  ClickableChipProps,
+  DeletableChipProps,
+} from 'src/molecules/Chip/Chip';
 import { InteractiveChipStyle } from 'src/molecules/Chip/Chip.styles';
 
-// Type for InteractiveChipStyle (button)
-export type InteractiveChipProps = BaseChipProps &
-  HTMLAttributes<HTMLButtonElement>;
+export type InteractiveChipProps = (ClickableChipProps | DeletableChipProps) &
+  Omit<HTMLAttributes<HTMLButtonElement>, 'onClick'>;
 
 export const InteractiveChip = forwardRef<
   HTMLButtonElement,
   InteractiveChipProps
 >((props, ref) => {
   const {
     children,
-    onDelete,
     disabled = false,
     variant,
-    onClick,
     leadingIconData,
-    ...otherInteractiveProps
+    onClick,
+    onDelete,
+    ...otherProps
   } = props;
-
-  // deletable and clickable logic can be handled within this component
-  const deletable = onDelete !== undefined;
-  const clickable = !disabled && props.onClick !== undefined;
-  const handleDelete = disabled ? undefined : onDelete;
-  const handleClick = disabled ? undefined : onClick;
+  const relevantOnAction = onDelete ? onDelete : onClick;
 
   const handleKeyPress = (event: KeyboardEvent<HTMLButtonElement>) => {
     const { key } = event;
     if (key === 'Enter') {
-      if (deletable) {
-        handleDelete?.(event); // using non-null assertion operator since we know deletable is true
-      } else if (clickable) {
-        handleClick?.(event); // using non-null assertion operator since we know clickable is true
+      if ('onClick' in props) {
+        relevantOnAction(event);
       }
```

Comment:
If you focus a deletable chip, and press enter, i dont think onDelete will be called, but i think it should be

Reply:
New commit